### ISSUE TITLE
Unsatisfactory performances from compounds as weakly-supervised label

### ISSUE THREAD

**Issue by Willaem** [crosszamirski/WS-DINO#4]
Hi,

I have been trying to achieve results similar to the ones published in your paper using compounds as weak labels, however, I am unable to reach the expected NSCB. The values obtained for per-channel NSCB are, for channels 0, 1, 2 and combined : 

(0.7282608695652174, 0.8043478260869565, 0.5978260869565217, 0.9021739130434783)

These are results from 400 epochs of training, using the exact same code, parameters and architecture (ViT-Small), except for the GPU used during training, which is a NVidia V100SXM2 (16G memory). 

These would align with the similar issue presented [here](https://github.com/crosszamirski/WS-DINO/issues/2). Is it possible that the expected NSCB of 0.96 is only achievable on a specific seed?

Furthermore, the training logs for weak compounds (NSC_NSCB_Weak_Compound.csv) included in the code stop at epoch 280. Do you train the model further, as you explain in the paper? Would it be possible to have access to the final computed features for the training set or the rest of the training logs?

Thank you for your answers,
William Leclerc

**Comment by crosszamirski** [comment]
Hi William,

400 epochs is the parameter used as the total number of epochs which sets the adaptive values in training, however every model we trained would reach peak performance at more like 200 epochs, sometimes less. Training to 400 epochs will certainly result in worse performance. Have you checked NSC and NSCB values for earlier epochs (i.e. prior to overfitting?)

**Comment by LiuBodan** [comment]
Hi William @Willaem,

Did you reproduce the result sccessfully eventually? I'm having similar issues for the reproduction and I'm using a similar GPU as yours.

Thanks
